tools/flang1/flang1exe/semfunc.c

```diff
@@ -1247,8 +1248,39 @@ func_call2(SST *stktop, ITEM *list, int flag)
     goto exit_;
   }
   ii = 0;
-  /* before processing arguments, add derived type return values if needed */
-  argt = mk_argt(argt_count); /* mk_argt stuffs away count */
+  /* As for assumed-length character function, get a temporary with DT_DEFERCHAR, SC_BASED here */
+  if ((DTY(dtype) == TY_CHAR) && ADJLENG(func_sptr)) {
+    int functmp;
+    int funcast, alloc_ast, len_ast;
+    int alloc_stmt, len_stmt;
+
+    functmp = getcctmp_sc('d', sem.dtemps++, ST_VAR, DT_DEFERCHAR, SC_BASED);
+    funcast = mk_id(functmp);
+    get_static_descriptor(functmp);
+    get_all_descriptors(functmp);
+    ALLOCDESCP(functmp, 1);
+    ALLOCP(functmp, 1);
+    len_ast = DTY(dtype + 1);
+    len_stmt = mk_assn_stmt(get_len_of_deferchar_ast(funcast), len_ast, A_DTYPEG(len_ast));
+    /* Allocate the temporary */
+    alloc_ast = mk_stmt(A_ALLOC, 0);
+    A_TKNP(alloc_ast, TK_ALLOCATE);
+    A_SRCP(alloc_ast, funcast);
+    alloc_stmt = add_stmt(alloc_ast);
+    sem.alloc_std = alloc_stmt;
+    add_stmt_before(len_stmt, alloc_stmt);
+    add_auto_dealloc(functmp);
+    /* add the temporary and length of character to ARGT list */
+    return_value = funcast;
+    argt_count = argt_count + 2;
+    argt = mk_argt(argt_count);
+    ARGT_ARG(argt, 0) = return_value;
+    ARGT_ARG(argt, 1) = mk_convert(len_ast, stb.user.dt_int);
+    ii = ii + 2;
+  } else {
+    /* before processing arguments, add derived type return values if needed */
```

Comment:
Is this comment still valid? This block doesn't seem to handle derived type return values.

Reply:
> Is this comment still valid? This block doesn't seem to handle derived type return values.

Indeed, this block does not handle derived type return values, I think this comment was added as a hint that if a derived type return value exists, it should be added to the ARGT list.